tests/vfs_open_mode_tests.cpp

```diff
@@ -0,0 +1,99 @@
+#include <sqlite_orm/sqlite_orm.h>
+#include <catch2/catch_all.hpp>
+#include <cstdio>  //  std::remove
+#if SQLITE_ORM_HAS_INCLUDE(<filesystem>)
+#include <filesystem>
+#endif
+#include "catch_matchers.h"
+
+#ifdef SQLITE_ORM_CTAD_SUPPORTED
+
+struct User {
+    std::string id;
+};
+
+using namespace sqlite_orm;
+
+static const auto default_table = make_table("users", make_column("id", &User::id, primary_key()));
+
+TEST_CASE("vfs modes open successfully") {
+
+#if defined(SQLITE_ORM_APPLE)
+    internal::string_constant_type vfs =
```

Comment:
can we use `std::string_view` instead to avoid `const char*` usage? Just want to keep OOP style and to skip C strings where possible

Reply:
no, not yet. not all STL libraries are C++17